src/IndexMaps/realindexmap.jl

```diff
@@ -75,12 +76,115 @@ function calculate_ind_values(imap::RealIndexMap, xs::Vector, dims::Vector{Int})
   return ind_to_ind_value_map
 end
 
-function grid_points(imap::RealIndexMap, N::Int, d::Int)
+""" 
+    grid_points(imap, N, d, span; exact_grid=true, enforced=[])
+
+  Gives `N` grid points from a given dimension of `imap` within a specified range.
+
+  # Arguments
+  - `imap::RealIndexMap`: An IndexMap specifying the structure of the TN being used
+  - `N::Int`: The number of grid points requested.
+  - `d::Int` The index of the dimension of `imap` requested
+  - `span::AbstractVector{<:Number}`: A two element number vector [a,b] with 0≤a<b≤1. The right endpoint of this span is not included as a gridpoint in the output.
+  - `exact_grid::Bool`: Flag specifying whether the function should give exact grid points 
+  (note: using `exact_grid=true` may cause less than `N` grid points to be returned)
+  - `enforced::AbstractVector{<:Number}`: A list of points that we want to enforce to show up in the grid.
+"""
+function grid_points(
+  imap::RealIndexMap,
+  N::Int,
+  d::Int;
+  span::AbstractVector{<:Number}=[0, 1],
+  exact_grid::Bool=true,
+  enforced=[],
+)
+  if length(span) != 2 || span[1] >= span[2] || span[1] < 0 || span[2] > 1
+    throw(
+      "expected a two-element vector [a,b] with 0≤a<b≤1 as input span, instead found $span"
+    )
+  end
+
+  dims = dim.(dimension_inds(imap, d))
+  @assert all(y -> y == first(dims), dims)
+  base = first(dims)
+  L = length(dimension_inds(imap, d))
+
+  if !exact_grid
+    grid_points = range(span[1], span[2] - (span[2] - span[1]) / N; length=N)
+  else #exact_grid = true
```

Comment:
I am think this function should follow some more straightforward logic to simplify this code (even if its not quite optimal):

1. Generate an inexact grid using the `range` function
2. If `exact_grid = true` Round all the results to the nearest decimal compatible with the spacing `2^-L` (write a helper function for this) and return  only the unique values you get.

Reply:
I just made these changes in the simplify grid_points commit